librz/analysis/serialize_analysis.c

```diff
@@ -827,7 +834,13 @@ RZ_API RZ_NULLABLE RzAnalysisVar *rz_serialize_analysis_var_load(RZ_NONNULL RzAn
 	if (!name || !type || kind == -1 || delta == ST64_MAX) {
 		goto beach;
 	}
-	ret = rz_analysis_function_set_var(fcn, delta, kind, type, 0, arg, name);
+	char *error_msg = NULL;
+	RzType *vartype = rz_type_parse_string_single(fcn->analysis->typedb->parser, type, &error_msg);
+	if (error_msg) {
+		eprintf("Fail to parse the function variable (\"%s\") type: %s\n", name, type);
+		goto beach;
```

Comment:
This case should not fail the entire loading of the variable, since there might be cases where a variable type was "valid" before but is not anymore now and you might lose some valuable vars from your project.

What it should do instead would be to try to create some stub type like either a generic integer or void one or `RZ_TYPE_KIND_IDENTIFIER` with the identifier being the broken string so we can reconstruct the variable as good as possible.
Then the warning should not be emitted with `eprintf()` but through the `RzSerializeResultInfo *` (see other `RZ_SERIALIZE_ERR()` usages), which you probably have to pass in from the outside first, which means that the user will see the warning explicitly as part of the loading process.

We should also add a test for this best-possible recovery logic.

Reply:
Please open an issue. I will address this once this PR is merged.